File: dashboard/src/models/StockInformation.ts
export interface StockInformation {
  id: string
  ticker: string
  target_from: string
  target_to: string
  company: string
  action: string
  brokerage: string
  rating_from: string
  rating_to: string
  time: string
}

export interface StockWithScore {
    stock_information: StockInformation
    score: number
}

export interface StockQueryParams {
  company?: string
  brokerage?: string
  ticker?: string
  action?: string
  sort_by?: string
  order?: 'ASC' | 'DESC'
  limit?: number
  offset?: number
}

File: backend/models/stock-information.go
package models

type StockInformation struct {
	ID         string `json:"id"`
	Ticker     string `json:"ticker"`
	TargetFrom string `json:"target_from"`
	TargetTo   string `json:"target_to"`
	Company    string `json:"company"`
	Action     string `json:"action"`
	Brokerage  string `json:"brokerage"`
	RatingFrom string `json:"rating_from"`
	RatingTo   string `json:"rating_to"`
	Time       string `json:"time"`
}

type StockWithScore struct {
	StockInformation `json:"stock_information"`
	Score            float64 `json:"score"`
}
